bevy_lint/README.md

```diff
@@ -1,9 +1,168 @@
+
+Once you have the toolchain installed, you can compile and install `bevy_lint` through `cargo`:
+
+```bash
+rustup run $TOOLCHAIN_VERSION cargo install \    
+    --git https://github.com/TheBevyFlock/bevy_cli.git \
+    --tag $TAG \
+    --locked \
+    bevy_lint
+```
+
+Make sure to replace `$TOOLCHAIN_VERSION` and `$TAG` in the above command. The tag for a specific release can be found in the [releases tab](https://github.com/TheBevyFlock/bevy_cli/releases). For example, the tag for 0.1.0 is `lint-v0.1.0`.
+
+## Usage
+
+`bevy_lint` has the same API as the `cargo check` command:
+
+```bash
+bevy_lint --help
+```
+
+If you have the Bevy CLI installed, the linter is also available through the `lint` subcommand:
+
+```bash
+bevy lint --help
+```
+
+<div class="rustdoc-alert rustdoc-alert-note">
+
+> **Note**
+>
+> `bevy_lint` checks your code with the nightly toolchain it was installed with, meaning you _do_ have access to unstable features when it is called. This is best used when [detecting `bevy_lint`](#detecting-bevy_lint).
+
+</div>
+
+### Detecting `bevy_lint`
+
+The linter passes `--cfg bevy_lint` when it checks your code, allowing you to detect it:
+
+```rust,ignore
+// Conditionally include this function only when `bevy_lint` is used.
+#[cfg(bevy_lint)]
+fn foo() {
+    // ...
+}
+
+// Conditionally add an attribute only when `bevy_lint` is used.
+#[cfg_attr(bevy_lint, ...)]
+struct Foo;
+```
+
+If you use this, you may also need to register `bevy_lint` as a valid `cfg` flag in your `Cargo.toml`:
+
+```toml
+[lints.rust]
+unexpected_cfg = { level = "warn", check-cfg = ["cfg(bevy_lint)"] }
+```
+
+### Registering `bevy` as a Tool
+
+When you run `bevy_lint` on a project, `rustc` knows an exact list of all `bevy::` lints registered. With this it can detect that `bevy::missing_reflect` is valid and `bevy::uh_oh` isn't, and emit a corresponding warning.
+
+When you run normal `cargo check`, however, it does not know about _any_ `bevy::` lints. In order to avoid erroring on _all_ usages of `bevy::`, but to still provide good diagnostics on typos, the `#![register_tool(...)]` attribute was introduced.
+
+```rust,ignore
+// Note that this is nightly-only. We'll get to that in a second!
+#![register_tool(bevy)]
+```
+
+Using `#![register_tool(bevy)]` tells the compiler that `bevy` is a valid name in attributes, even if it does not know what `bevy` is.[^rustfmt-skip] When `cargo check` now runs over a project with `#[warn(bevy::lint_name)]`, it will simply skip it instead of emitting an error. (But running `bevy_lint` will still detect and check this attribute as normal.)
+
+[^rustfmt-skip]: If you've ever used `#[rustfmt::skip]` in your code, this is how `rustc` avoids erroring on it. However unlike the `bevy` namespace, `rustfmt` is registered automatically without a need for `#![register_tool(rustfmt)]` due to it being an official tool.
```

Comment:
Neat!

Reply:
Yup! If your curious, you can find the list of predefined tools in [`rustc_resolve::macros`](https://github.com/rust-lang/rust/blob/80445576d04c989ff011e152f026f613f8493a21/compiler/rustc_resolve/src/macros.rs#L152-L155).